### ISSUE TITLE
Add backpressure for SDK.capture methods

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#4423]
### Description

A customer reported that they called `SentrySDK.captureException` in a tight loop to report failed HTTP requests, and the watchdog terminated their app due to OOM. Not only `captureException`, but also `captureError`, `captureMessage` are expensive operations that can cause trouble when being called in a tight loop. We can implement a backpressure mechanism to avoid apps being terminated by the watchdog.

This is related to https://github.com/getsentry/sentry-cocoa/issues/4038.

**Comment by philprime** [comment]
@philipphofmann will this be solved with the [Telemetry Processor](https://develop.sentry.dev/sdk/telemetry/telemetry-processor/)?

**Comment by philippsuess** [comment]
No, the telemetry processor won't fix that. Ideally, the backpressure mechanism should already drop data before the client. I think we can close this for now. This hasn't surfaced since I created the issue 1,5 years ago. We can open a new one if this comes up again.
